File: Lunchtime/Controller/SetupPageFifthViewController.m
//
//  SetupPageSecondViewController.m
//  Lunchtime
//
//  Created by Willow Belle on 2015-11-13.
//  Copyright © 2015 Cosmic Labs. All rights reserved.
//

#import "SetupPageFifthViewController.h"
#import "User.h"
#import "LunchtimeNotification.h"

@interface SetupPageFifthViewController ()

@property (nonatomic, weak) IBOutlet UIDatePicker *lunchtimeDatePicker;

@end

@implementation SetupPageFifthViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    [self.lunchtimeDatePicker setDatePickerMode:UIDatePickerModeTime];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];

    RLMRealm *realm = [RLMRealm defaultRealm];
    User *user = [User new];
    
    [user setPreferredDistance:@1500];
    [user setLunchtime:self.lunchtimeDatePicker.date];
    [user setIdentifier:@1];

    [realm beginWriteTransaction];
    [User createOrUpdateInRealm:realm withValue:user];
    [realm commitWriteTransaction];

    // Setup and schedule the lunchtime notification
    LunchtimeNotification *notification = [LunchtimeNotification lunchtimeNotificationWithDate:self.lunchtimeDatePicker.date];
    [[UIApplication sharedApplication] scheduleLocalNotification:notification];
    NSLog(@"firedate = %@", notification.fireDate);
}

@end

File: Lunchtime/Controller/SetupPageFifthViewController.swift
//
//  SetupPageFifthViewController.swift
//  Lunchtime
//
//  Created by Willow Belle on 2015-11-18.
//  Copyright © 2015 Cosmic Labs. All rights reserved.
//

import UIKit
import Realm

class SetupPageFifthViewController: UIViewController {

    // MARK: - Outlets
    @IBOutlet weak var picker: UIDatePicker?

    // MARK: - Controller Lifecycle
    override func viewDidLoad() {
        super.viewDidLoad()

        picker?.datePickerMode = .Time
    }

    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)

        guard let picker = picker else {
            return
        }
        
        let realm = RLMRealm.defaultRealm()
        let user = User()

        user.lunchtime = picker.date
        user.preferredDistance = NSNumber(integer: 1500)
        user.identifier = NSNumber(integer: 1)
        user.category = "food"

        do {
            realm.beginWriteTransaction()
            User.createOrUpdateInRealm(realm, withValue: user)
            try realm.commitWriteTransaction()
        } catch let error {
            print("Something went wrong...")
            print("Error Reference: \(error)")
        }

        let notification = LunchtimeNotification(date: picker.date)
        UIApplication.sharedApplication().scheduleLocalNotification(notification)
    }
}
